src/pages/WorkspaceDetails/Header/Actions/__tests__/Actions.spec.tsx

```diff
@@ -10,132 +10,162 @@
+import { createHashHistory, History } from 'history';
 import React from 'react';
+import { Provider } from 'react-redux';
 import { HeaderActionSelect } from '../';
 import { WorkspaceAction, WorkspaceStatus } from '../../../../../services/helpers/types';
-
-jest.mock('../../../../../components/Workspace/DeleteAction', () => {
-  const FakeWorkspaceDeleteAction = React.forwardRef((props, ref) => (
-    <button ref={ref as any}></button>
-  ));
-  FakeWorkspaceDeleteAction.displayName = 'WorkspaceDeleteAction';
-
-  return FakeWorkspaceDeleteAction;
+import { AppThunk } from '../../../../../store';
+import { ActionCreators, ResourceQueryParams } from '../../../../../store/Workspaces';
+import { FakeStoreBuilder } from '../../../../../store/__mocks__/storeBuilder';
+
+jest.mock('../../../../../store/Workspaces/index', () => {
+  return {
+    actionCreators: {
+      startWorkspace: (workspaceId: string, params?: ResourceQueryParams): AppThunk<any, Promise<void>> => async (): Promise<void> => {
+        return Promise.resolve();
+      },
+      stopWorkspace: (workspaceId: string): AppThunk<any, Promise<void>> => async (): Promise<void> => {
+        return Promise.resolve();
+      }
+    } as ActionCreators,
+  };
 });
 
-describe('Workspace WorkspaceAction widget', () => {
-  const workspaceName = 'test-workspace-name';
-  const workspaceId = 'test-workspace-id';
-  const workspaceStatus = WorkspaceStatus.STOPPED;
+const workspaceName = 'test-workspace-name';
+const workspaceId = 'test-workspace-id';
+const workspaceStoppedStatus = WorkspaceStatus.STOPPED;
+const store = new FakeStoreBuilder().withWorkspaces({
+  workspaceId,
+  workspaceName,
+  workspaces: [{
+    id: workspaceId,
+    namespace: 'test',
+    status: WorkspaceStatus[workspaceStoppedStatus],
+    devfile: {
+      apiVersion: 'v1',
+      components: [],
+      metadata: {
+        name: workspaceName,
+      }
+    }
+  }]
+}).build();
 
-  it('should call the callback with OPEN action', () => {
+describe('Workspace WorkspaceAction widget', () => {
+  it('should call the callback with OPEN action', async () => {
     const action = WorkspaceAction.OPEN_IDE;
-    const onAction = jest.fn();
-    const component = createComponent(workspaceStatus, workspaceName, workspaceId, onAction);
+    const history = createHashHistory();
+    const component = createComponent(workspaceStoppedStatus, workspaceName, workspaceId, history);
 
     renderComponent(component);
 
     const actionDropdown = screen.getByTestId(`${workspaceId}-action-dropdown`);
     actionDropdown.click();
 
-    expect(onAction).not.toBeCalled();
+    expect(history.location.pathname).toBe('/');
 
     const targetAction = screen.getByText(action);
     targetAction.click();
 
-    expect(onAction).toBeCalledWith(action);
+    await waitFor(() => history.location.pathname === '/ide/test/test-workspace-name');
+    expect(history.location.pathname).toBe('/ide/test/test-workspace-name');
```

Comment:
I suppose this should work:

```suggestion
    await waitFor(() => expect(history.location.pathname).toBe('/ide/test/test-workspace-name'));
```